ludwig/backend/ray.py

```diff
@@ -303,15 +282,65 @@ def tune_learning_rate_fn(
         hvd.shutdown()
 
 
+# @DeveloperAPI
+# class TqdmCallback(rt.TrainingCallback):
+#     """Class for a custom ray callback that updates tqdm progress bars in the driver process."""
+
+#     def __init__(self) -> None:
+#         """Constructor for TqdmCallback."""
+#         super().__init__()
+#         self.progess_bars = {}
+
+#     def process_results(self, results: List[Dict], **info) -> None:
+#         """Called everytime ray.train.report is called from subprocesses. See
+#         https://docs.ray.io/en/latest/train/api.html#trainingcallback.
+
+#         # Inputs
+
+#         :param results: (List[Dict]) List of results from the training function.
+#             Each value in the list corresponds to the output of the training function from each worker.
+
+#         # Return
+
+#         :return: (None) `None`
+#         """
+#         for result in results:
+#             progress_bar_opts = result.get("progress_bar")
+#             if not progress_bar_opts:
+#                 continue
+#             # Skip commands received by non-coordinators
+#             if not progress_bar_opts["is_coordinator"]:
+#                 continue
+#             _id = progress_bar_opts["id"]
+#             action = progress_bar_opts.pop("action")
+#             if action == "create":
+#                 progress_bar_config = progress_bar_opts.get("config")
+#                 self.progess_bars[_id] = tqdm.tqdm(**progress_bar_config)
+#             elif action == "close":
+#                 self.progess_bars[_id].close()
+#             elif action == "update":
+#                 update_by = progress_bar_opts.pop("update_by")
+#                 self.progess_bars[_id].update(update_by)
+
+
 @DeveloperAPI
-class TqdmCallback(rt.TrainingCallback):
+class TqdmCallback(ray.tune.Callback):
     """Class for a custom ray callback that updates tqdm progress bars in the driver process."""
 
     def __init__(self) -> None:
         """Constructor for TqdmCallback."""
         super().__init__()
         self.progess_bars = {}
 
+    def on_trial_result(iteration: int, trials: List["Trial"], trial: "Trial", result: Dict, **info):
+        print("On_Trial_Result")
+        print(f">>> iteration: {iteration}")
+        print(f">>> trials: {trials}")
+        print(f">>> trial: {trial}")
+        print(f">>> result: {result}")
+        print(f">>> info: {info}")
```

Comment:
Seems like this is called everytime `session.report()` is called in subprocesses.

Reply:
Have a fix that's working locally, will push in a moment